text/3502-cargo-script.md

```diff
@@ -0,0 +1,1205 @@
+- Feature Name: `cargo-script`
+- Start Date: 2023-03-31
+- Pre-RFC: [internals](https://internals.rust-lang.org/t/pre-rfc-cargo-script-for-everyone/18639)
+- eRFC PR: [rust-lang/rfcs#3424](https://github.com/rust-lang/rfcs/pull/3424)
+- RFC PR: [rust-lang/rfcs#3502](https://github.com/rust-lang/rfcs/pull/3502)
+- Rust Issue: [rust-lang/cargo#12207](https://github.com/rust-lang/cargo/issues/12207)
+
+# Summary
+[summary]: #summary
+
+This RFC adds support for single-file bin packages in cargo.
+Single-file bin packages are `.rs` files with an embedded manifest and a
+`main`.
+These files will be accepted by cargo commands as `--manifest-path` just like `Cargo.toml` files.
+`cargo` will be modified to accept `cargo <file>.rs` as a shortcut to `cargo
```

Comment:
Given the example doesn't use the `.rs` extension that should probably be denoted as optional here.

Reply:
This was meant to focus on the semantics of `.rs` files and not the literal extension, leaving discussion of that for later.  I've updated it to just say "rust source files" to avoid people reading too much into it.